### ISSUE TITLE
Spearhead: Dark Settings on mobile shows site's text color as dark over dark background

### ISSUE THREAD

**Issue by kspilarski** [Automattic/themes#2819]
## Steps to replicate

- activate Spearhead theme 
- in your mobile device, enable Dark appearance then visit the site (in mobile)
- then click menu and the texts will be dark over dark background: 

<img width="312" alt="Screen Shot 2020-11-15 at 5 14 24 PM" src="https://user-images.githubusercontent.com/36863633/99190391-a3039300-2766-11eb-95de-b830562f81e4.png">

For comparison, here's how Light appearance in mobile settings display the menu text color: 

<img width="309" alt="Screen Shot 2020-11-15 at 5 20 27 PM" src="https://user-images.githubusercontent.com/36863633/99190466-f544b400-2766-11eb-9e96-441b985c42a5.png">

## Result

Selecting dark appearance in mobile device and visiting the site will show the menu text color in dark color over dark background, making the texts hard to read. 

## Expected

Dark appearance in mobile device setting should set the background dark, but the menu text color should be white. 

* User report: 3490288-zd
* Follow-up ticket: 3490288-zd

**Comment by danieldudzic** [comment]
Hi @kspilarski !

I was unable to replicate the issue. Would you mind testing this again?

Please let me know.

**Comment by scruffian** [comment]
I can replicate it, if you choose the first two palettes here: 
<img width="291" alt="Screenshot 2021-01-15 at 10 00 14" src="https://user-images.githubusercontent.com/275961/104710849-793ffb80-5718-11eb-80e6-b3f88d9527de.png">

**Comment by danieldudzic** [comment]
@scruffian Thank you for having a look.

I believe I have misunderstood the issue originally. Will have another look today.

**Comment by scruffian** [comment]
We discussed this on Slack and decided that the best way forward would be to add a class to the body when a site has selected custom colors. Then we can use that class to remove the dark mode when color palettes are active.

**Comment by pbking** [comment]
I think we might have an issue doing the class-to-limit-darkmode.  It sounds like it would be the cleanest, however I think I see the following problem:

We could do something like the following for the dark mode variable setting:
```
@media ( prefers-color-scheme: dark ) {
	body:not(.custom-colors) {
		--global--color-primary: #ff6a81;
```

That’s[ instead of using](https://github.com/Automattic/themes/blob/trunk/spearhead/variables.css#L106) `:root`.  This will cause those dark mode variables to be set on any element from the body down, but only if the ‘.custom-colors’ class isn’t set on that body.

(NOTE: I also didn’t understand logic to the addition/removal .custom-colors class. Another class would have to be added/removed instead for this purpose but that class works for demonstration purposes.)

So far that works out.  However in Seedlet’s variable handling there are a few variables that reference other variables.  [Among them](https://github.com/Automattic/themes/blob/trunk/seedlet/assets/sass/child-theme/variables.css#L202):
`--entry-header--color: var(--global--color-primary);`

Those variables are scoped to :root, therefore the value of `—entry-header—color` is the value of `--global--color-primary` that is scoped to :root (which, if done as above, is the light color scheme even if dark-mode media query checks out because those dark mode variable changes are scoped to a class, not :root)

So in order for that method to work Spearhead would have to re-assign all of the variables that are referencing any of the color variables that are being overridden by dark mode color variables.  (Do-able but with a code smell.) 

There are a few ways to dance those variables around, but if it all comes down to using <anything> to limit the scope of the assignment of those variables (other than a media query) then we wind up with the same problem.

**Comment by jffng** [comment]
Closed by #3105, thanks for reporting!
